### ISSUE TITLE
Backport 987087df9002 to release/18.x

### ISSUE THREAD

**Issue by SixWeining** [llvm/llvm-project#79253]
/cherry-pick baba7e4175b6ca21e83b1cf8229f29dbba02e979

This can address buildbot failures:
-    http://45.33.8.238/macm1/77360/step_11.txt
-    http://45.33.8.238/linux/128902/step_12.txt

**Comment by SixWeining** [comment]
Hi @tstellar, I know release/18.x is just created. May I cherry-pick this fix into 18.x?

**Comment by nikic** [comment]
Failed workflow: https://github.com/llvm/llvm-project/actions/runs/7635785866/job/20801796617 I think it's because empty `--phab-token` eats the next arg.

**Comment by SixWeining** [comment]
> Failed workflow: https://github.com/llvm/llvm-project/actions/runs/7635785866/job/20801796617 I think it's because empty `--phab-token` eats the next arg.

I agree. And thanks for the fix.

**Comment by nikic** [comment]
Now it fails with (https://github.com/llvm/llvm-project/actions/runs/7638782349/job/20810392396):
```
cherry-picking ['baba7e4175b6ca21e83b1cf8229f29dbba02e979']
Pushing to https://***@github.com/llvmbot/llvm-project issue79253
Traceback (most recent call last):
  File "/home/runner/work/llvm-project/llvm-project/./llvm/utils/git/github-automation.py", line 679, in <module>
    parser.add_argument(
  File "/home/runner/work/llvm-project/llvm-project/./llvm/utils/git/github-automation.py", line 578, in execute_command
    https://docs.github.com/en/get-started/quickstart/github-glossary#compare-branch
  File "/home/runner/work/llvm-project/llvm-project/./llvm/utils/git/github-automation.py", line 475, in create_branch
    ) -> github.IssueComment.IssueComment:
  File "/home/runner/work/llvm-project/llvm-project/./llvm/utils/git/github-automation.py", line 356, in issue_notify_branch
    based on the text in stdin.
  File "/home/runner/.local/lib/python3.10/site-packages/github/Issue.py", line 343, in create_comment
    headers, data = self._requester.requestJsonAndCheck(
  File "/home/runner/.local/lib/python3.10/site-packages/github/Requester.py", line 442, in requestJsonAndCheck
    return self.__check(
  File "/home/runner/.local/lib/python3.10/site-packages/github/Requester.py", line 487, in __check
    raise self.__createException(status, responseHeaders, data)
github.GithubException.GithubException: 403 {"message": "Resource not accessible by personal access token", "documentation_url": "https://docs.github.com/rest/issues/comments#create-an-issue-comment"}
```

I think this is because the workflow is defined with `permissions: read`.

**Comment by nikic** [comment]
Note that the backport branch did get successfully created at https://github.com/llvmbot/llvm-project/tree/issue79253. It failed when posting the issue comment.

**Comment by tstellar** [comment]
Merged: c9e73cdd9a17f15ede120ea57657553f9e105eab
